src/day.rs

```diff
@@ -0,0 +1,231 @@
+    Tz: TimeZone + Copy + Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl<Tz> PartialOrd for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.date.partial_cmp(&other.date)
+    }
+}
+
+impl<Tz> Ord for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.date.cmp(&other.date)
+    }
+}
+
+impl<Tz> PartialEq for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.date.eq(&other.date)
+    }
+}
+
+impl<Tz> Eq for Day<Tz> where Tz: TimeZone + Copy + Display {}
+
+impl<Tz> Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    ///
+    pub fn date(&self) -> NaiveDate {
+        self.date
+    }
+
+    ///
+    pub fn zone(&self) -> Tz {
+        self.tz
+    }
+
+    ///
+    pub fn new(date: NaiveDate, tz: Tz) -> Day<Tz> {
+        Day { date, tz }
+    }
+
+    ///
+    pub fn succ(&self) -> Option<Day<Tz>> {
+        Some(Day { date: self.date.succ_opt()?, tz: self.tz })
+    }
+
+    ///
+    pub fn pred(&self) -> Option<Day<Tz>> {
+        Some(Day { date: self.date.pred_opt()?, tz: self.tz })
+    }
+
+    ///
+    pub fn start(&self) -> DateTime<Tz> {
+        // All possible offsets: https://en.wikipedia.org/wiki/List_of_UTC_offsets
+        // means a gap of 15 minutes should be reasonable
+
+        // while looping here is less than ideal, in the vast majority of cases
+        // the inital start time guess will be valid
+
+        let base = NaiveTime::MIN;
+        for multiple in 0..=24 {
+            let start_time = base + oldtime::Duration::minutes(multiple * 15);
+            match self.tz.from_local_datetime(&self.date.and_time(start_time)) {
+                crate::LocalResult::None => continue,
+                crate::LocalResult::Single(dt) => return dt,
+                crate::LocalResult::Ambiguous(dt1, dt2) => {
+                    if dt1.naive_utc() < dt2.naive_utc() {
+                        return dt1;
+                    } else {
+                        return dt2;
+                    }
+                }
+            }
+        }
+
+        panic!("Unable to calculate start time for date {} and time zone {}", self.date, self.tz)
+    }
+
+    ///
+    pub fn checked_add_days(self, days: Days) -> Option<Self> {
+        if days.0 == 0 {
+            return Some(self);
+        }
+
+        i64::try_from(days.0).ok().and_then(|d| self.diff_days(d))
+    }
+
+    ///
+    pub fn checked_sub_days(self, days: Days) -> Option<Self> {
+        if days.0 == 0 {
+            return Some(self);
+        }
+
+        i64::try_from(days.0).ok().and_then(|d| self.diff_days(-d))
+    }
+
+    fn diff_days(self, days: i64) -> Option<Self> {
+        let date = self.date.checked_add_signed(Duration::days(days))?;
+        Some(Day { date, ..self })
+    }
```

Comment:
I feel like in general we should start with a minimal API here, we can grow it over time as people actually ask for it.

Reply:
Agreed, however I added these specifically to use here https://github.com/esheppa/db-dump/blob/7350bfabb9ff48b999ba4d6b0e827e8e2eea275c/examples/industry-coefficient.rs#L37